File: JogoCSharp/Models/Heroi.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace JogoCSharp.Models
{
    public class Heroi
    {

        public string Nome { get; set; }
        public int Idade { get; set; }
        public string Tipo { get; set; }

        public Heroi(string nome, int idade, string tipo)
        {
            Nome = nome;
            Idade = idade;
            Tipo = tipo;

        }
        public string Atacar()
        {
            return Tipo.ToLower() switch
            {
                "mago" => $"o {Tipo} atacou usando magia",
                "guerreiro" => $"o {Tipo} atacou usando espada",
                "monge" => $"o {Tipo} atacou usando artes marciais",
                "ninja" => $"o {Tipo} atacou usando shuriken",
                _ => $"o {Tipo} atacou de forma desconhecida"
            };
        }
    }
}

File: ClasseDoHeroi/src/main/java/com/example/ClasseDoHeroi/model/Heroi.java
package com.example.ClasseDoHeroi.model;

public class Heroi {
    private String nome;
    private int idade;
    private String tipo;

    public Heroi(String nome, int idade, String tipo) {
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;
    }

    public String atacar() {
        switch (tipo.toLowerCase()) {
            case "mago": return "O mago atacou usando magia";
            case "guerreiro": return "O guerreiro atacou usando espada";
            case "monge": return "O monge atacou usando artes marciais";
            case "ninja": return "O ninja atacou usando shuriken";
            default: return "O herói não sabe atacar";
        }
    }
}
